shotover-proxy/src/transforms/protect/crypto.rs

```diff
@@ -58,9 +62,27 @@ pub async fn decrypt(
         .cached_get_key(key_id, Some(protected.enc_dek), Some(protected.kek_id))
         .await?;
 
-    let decrypted_bytes = secretbox::open(&protected.cipher, &protected.nonce, &sym_key.plaintext)
+    let nonce = Nonce::from_slice(&protected.nonce);
+    let cipher = ChaCha20Poly1305::new(Key::from_slice(&sym_key.plaintext));
+
+    let decrypted_bytes = cipher
+        .decrypt(nonce, &*protected.cipher)
         .map_err(|_| anyhow!("couldn't decrypt value"))?;
 
     //TODO make error handing better here - failure here indicates an authenticity failure
     bincode::deserialize(&decrypted_bytes).map_err(|_| anyhow!("couldn't decrypt value"))
 }
+
+pub fn gen_key() -> Key {
+    let mut key_bytes = [0; 32];
+    let mut rng = rand::thread_rng();
+    rng.fill_bytes(&mut key_bytes);
+    *Key::from_slice(&key_bytes)
+}
+
+pub fn gen_nonce() -> Nonce {
+    let mut rng = rand::thread_rng();
```

Comment:
We need to use a cryptographically secure random byte source here. Previously we called gen_nonce which would end up calling out to libsodiums random bytes function. Does RustCryptos ecosystem give us a better alternative

Reply:
huh.
rand 0.6.5 claims that thread_rng is cryptographically secure: https://docs.rs/rand/0.6.5/rand/rngs/struct.ThreadRng.html
However the latest rand 0.8.5 does not make any claims in regards to being cryptographically secure: https://docs.rs/rand/0.8.5/rand/rngs/struct.ThreadRng.html

Reply:
I would feel safer if we called a rustcrypto rand source, even if it was just a re-export of rand::thread_rng

safety in numbers and all that :)

Reply:
I can use`rand_core` because it's re-exported but not `rand`.